src/lit-element.ts

```diff
@@ -236,10 +232,12 @@ export class LitElement extends UpdatingElement {
   }
 
   /**
-   * Invoked on each update to perform rendering tasks. This method must return
-   * a lit-html TemplateResult. Setting properties inside this method will *not*
-   * trigger the element to update.
+   * Invoked on each update to perform rendering tasks. This method may return
+   * any value renderable by lit-html's NodePart - typically a TemplateResult.
+   * Setting properties inside this method will *not* trigger the element to
+   * update.
    */
-  protected render(): TemplateResult|void {
+  protected render(): unknown {
```

Comment:
Can you elide the return if you type this as `unknown|void`?

Reply:
Doesn't seem like it:
```
src/lit-element.ts:241:23 - error TS2355: A function whose declared type is neither 'void' nor 'any' must return a value.                                                                                         

241   protected render(): unknown|void {
                          ~~~~~~~~~~~~
```

Reply:
It's a shame to add non-observable, non-type annotation code to satisfy the type-checker. Can we cast our way out of this?

Reply:
I don't know how I would cast the function in place since class syntax isn't expression based (and I assume we don't want to bind it / remove it from the prototype by making it a property), so I tried pulling it out of the class as this:

```javascript
LitElement.prototype.render = function() {} as () => unknown;
```

But then I get another couple of errors that seem like TS is no longer recognizing that `render` is also on the prototype:

```
src/lit-element.ts:219:18 - error TS2576: Property 'render' is a static member of type 'LitElement'

219             this.render(),
                     ~~~~~~

src/lit-element.ts:243:22 - error TS2576: Property 'render' is a static member of type 'LitElement'

243 LitElement.prototype.render = function() {} as () => unknown;
                         ~~~~~~
```

I'm not familiar with pre-classes TS, so maybe that's not the right way to do it?

Reply:
(from offline) I'm going to leave the `return undefined;` as is and assume that any given minifier should be smart enough to remove it.